playwright/src/test/java/AllowedOriginsTest.java

```diff
@@ -0,0 +1,136 @@
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.Objects;
+import java.util.Scanner;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class AllowedOriginsTest {
+
+  private Page page;
+  private Browser browser;
+
+  private static String axeSource;
+  private String axeForceLegacy;
+
+  private String addr() {
+    return "http://localhost:1337";
+  }
+
+  @BeforeClass
+  public static void reloadSource() throws IOException {
+    URL oldSourceUrl = AxeBuilder.class.getResource("/axe.min.js");
+    axeSource = URLReader(oldSourceUrl, StandardCharsets.UTF_8);
+  }
+
+  private static String URLReader(URL url, Charset encoding) throws IOException {
+    String content;
+    try (Scanner scanner = new Scanner(url.openStream(), String.valueOf(encoding))) {
+      content = scanner.useDelimiter("\\A").next();
+    }
+    return content;
+  }
+
+  private String downloadFromURL(String url) throws Exception {
+    // https://stackoverflow.com/a/13632114
+    try (InputStream stream = new URL(url).openStream()) {
+      return new Scanner(stream, "UTF-8").useDelimiter("\\A").next();
+    }
+  }
+
+  private void appendAxeSourceWithString(String source) throws IOException, URISyntaxException {
+    URL axeUrl = AxeBuilder.class.getResource("/axe.min.js");
+    Files.write(Paths.get(axeUrl.toURI().getPath()), source.getBytes(), StandardOpenOption.APPEND);
+  }
+
+  private void overwriteAxeSourceWithString(String source) throws IOException, URISyntaxException {
+    URL axeUrl = AxeBuilder.class.getResource("/axe.min.js");
+    Files.write(Paths.get(axeUrl.toURI().getPath()), source.getBytes(), StandardOpenOption.WRITE);
+  }
+
+  private Object getAllowedOrigins() {
+    return page.evaluate("axe._audit.allowedOrigins");
+  }
+
+  @Before
+  public void setup() throws Exception {
+    Playwright playwright = Playwright.create();
+    browser = playwright.chromium().launch(new BrowserType.LaunchOptions().setHeadless(true));
+    axeSource =
+        URLReader(
+            Objects.requireNonNull(AxeBuilder.class.getResource("/axe.min.js")),
+            StandardCharsets.UTF_8);
+
+    axeForceLegacy = downloadFromURL(addr() + "/axe-force-legacy.js");
+
+    page = browser.newPage();
+  }
+
+  @After
+  public void teardown() throws IOException {
+    URL currentSource = AxeBuilder.class.getResource("/axe.min.js");
+    Files.write(Paths.get(currentSource.getPath()), axeSource.getBytes());
+
+    browser.close();
+  }
+
+  @Test
+  public void shouldNotSetWhenRunningRunPartialAndNotLegacyMode() {
+    page.navigate(addr() + "/index.html");
+    new AxeBuilder(page).analyze();
+    ArrayList<?> allowedOrigins = (ArrayList<?>) getAllowedOrigins();
+
+    assertEquals(allowedOrigins.get(0), addr());
```

Comment:
Will need to import `java.util.Objects`

```suggestion
    ArrayList<String> origins = new ArrayList<>();
    origins.add(attr());
    assertTrue(Objects.deepEquals(allowedOrigins, origins));
```